src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,75 @@
+package duke.task;
+
+import duke.util.DukeException;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.FormatStyle;
+
+public class Deadline extends Task {
+
+    private static final DateTimeFormatter DATE_SHORT_FORMATTER = DateTimeFormatter.ofPattern("dd/MM/yyyy");
+    private static final DateTimeFormatter DATE_MED_FORMATTER = DateTimeFormatter.ofLocalizedDate(FormatStyle.LONG);
+
+    private LocalDate by;
+
+    /**
+     * Initialises a Deadline.
+     *
+     * @param description description of Deadline
+     * @param by date of Deadline
+     */
+    public Deadline(String description, LocalDate by) {
+        super(description);
+        this.by = by;
+    }
+
+
+    private static void checkFormat(String formattedString) throws DukeException {
+        int byIndex = formattedString.indexOf("/by ");
+        if (byIndex == -1)
+            byIndex = formattedString.length();
+
+        String keyword = formattedString.split(" ", 2)[0];
+
+        if (!keyword.startsWith("deadline"))
+            throw new DukeException("I can't seem to find the deadline keyword");
+        else if (formattedString.length() <= 9 || formattedString.substring(9, byIndex).isEmpty())
+            throw new DukeException("the description of deadline cannot be empty");
+        else if (byIndex == formattedString.length() || formattedString.length() < byIndex + 5)
+            throw new DukeException("the [/by] time of deadline cannot be empty");
```

Comment:
I like the way you check for the keywords and handle all the different exceptions. Perhaps you can add Egyptian braces for each if/elseif block to be consistent with the coding standard.